### DISCUSSION TITLE
Ability to disable generation of the 404 page

### DISCUSSION THREAD

**Discussion by raRaRa** [vercel/next.js#27881] [Ideas]
### Describe the feature you'd like to request

I want to have an option in the `next.config.js` file that can disable the generation of the 404 page. The main reason for this is because Cloudflare Pages assumes the app isn't SPA if it finds a 404.html in the root, as described here: https://developers.cloudflare.com/pages/platform/serving-pages

> If your project doesn't include a top-level 404.html file, Pages assumes that you're deploying a single-page application. This includes frameworks like React, Vue, and Angular. Pages' default single-page application behavior matches all incoming paths to the root (/), allowing you to capture URLs like /about or /help and respond to them from within your SPA.

But since Next.js always generates a 404.html in the root, accessing dynamic routes directly in Cloudflare Pages fails with a 404.

### Describe the solution you'd like

I want an option in `next.config.js` such as `disable404: true`, that disables the generation of the 404 page.

### Describe alternatives you've considered

Remove the 404.html file afterwards in the build process, not ideal.

**Comment by yunsii** [comment]
Still wait...

**Reply by raRaRa** [reply]
If you're also encountering this issue with Cloudflare Pages then I highly recommend poking them about adding the ability to change the SPA behavior. I've requested it many times but currently it looks like I'm the only one :(

**Reply by yunsii** [reply]
I define script `"rm404": "rimraf ./out/404 && rimraf ./out/404.html"` to remove 404 page for now.

**Comment by j2em6h** [comment]
I have this same request, I use a Next static build for select pages alongside a different statically generated site, and that other site handles the 404 pages. I do not need a 404 page generated by Next, I would love the ability to disable it. For now I just skip deploying the 404 files specifically, but would love to not have to handle them at all.
